core/src/test/scala/org/apache/spark/storage/MemoryStoreSuite.scala

```diff
@@ -546,4 +550,105 @@ class MemoryStoreSuite
       }
     }
   }
+
+  test("put native objects to MemoryStore and remove") {
```

Comment:
This seems quite complex - is there any way to test this more easily? If it's this hard to test directly, I may say just punt on it. This is kind of best-effort to close the objects anyway.

Reply:
@srowen, I see, I changed the UT as a dummy one, using one allocated counter to verify close functions were called after remove() or clear()